<sdf version='1.7'>
  <world name='default'>
    <plugin name="set_entity_state" filename="libgazebo_ros_state.so"/>
    <scene>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>1</shadows>
      <sky>
        <clouds>
          <speed>12</speed>
        </clouds>
      </sky>
      <fog>
        <type>linear</type>
        <start>10</start>
        <end>500</end>
        <density>1</density>
      </fog>
    </scene>

    <include>
      <uri>model://ground_plane</uri>
    </include>

    <include>
      <uri>model://sun</uri>
    </include>

    <model name="building_0">
      <include>
        <uri>model://building</uri>
        <pose>-5 -30 0 0 0 0</pose>
        <name>building_0</name>
      </include>
    </model>
    <model name="building_1">
      <include>
        <uri>model://building</uri>
        <pose>25 -30 0 0 0 0</pose>
        <name>building_1</name>
      </include>
    </model>
    <model name="building_2">
      <include>
        <uri>model://building</uri>
        <pose>5 30 0 0 0 0</pose>
        <name>building_2</name>
      </include>
    </model>
    <model name="building_3">
      <include>
        <uri>model://building</uri>
        <pose>-25 30 0 0 0 0</pose>
        <name>building_3</name>
      </include>
    </model>

    <model name="laser_0">
      <pose>0.0 0.0 1.0 0.0 0.0 0.0</pose>
      <link name="ray_link">
        <visual name="visual_box">
          <geometry>
            <box>
              <size>0.2 .1 .1</size>
            </box>
          </geometry>
        </visual>
        <collision name="collision_box">
          <geometry>
            <box>
              <size>0.02 .01 .01</size>
            </box>
          </geometry>
          <laser_retro>0.0</laser_retro>
        </collision>

        <!-- ray sensor -->
        <sensor name="sensor_ray" type="ray">
          <ray>
            <scan>
              <horizontal>
                <samples>16</samples>
                <resolution>1</resolution>
                <min_angle>0</min_angle>
                <max_angle>0</max_angle>
              </horizontal>
              <vertical>
                <samples>1</samples>
                <resolution>1</resolution>
                <min_angle>0.8</min_angle>
                <max_angle>0.8</max_angle>
              </vertical>
            </scan>
            <range>
              <min>0.05</min>
              <max>50.0</max>
            </range>
          </ray>
          <always_on>true</always_on>
          <update_rate>1.0</update_rate>
          <plugin name="laserscan" filename="libMultipathSensorPlugin.so">
            <ros>
              <namespace>/multipath</namespace>
              <remapping>~/out:=scan</remapping>
            </ros>
            <output_type>sensor_msgs/LaserScan</output_type>
            <disableNoise>0</disableNoise>
            <satNum>8</satNum>
            <satAzimuth>1.57 4.71 1.57 4.71 1.57 4.71 1.57 4.71</satAzimuth>
            <satElevation>0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0</satElevation>
          </plugin>
        </sensor>
      </link>
    </model>

<!-- 
    <gui fullscreen='0'>
      <camera name='user_camera'>
        <pose>-9.6407 49.4187 41.7671 0 0.6058 -1.55978</pose>
        <view_controller>orbit</view_controller>
        <projection_type>perspective</projection_type>
      </camera>
    </gui> -->

  </world>
</sdf>
<!-- vim: set et fenc=utf-8 ff=unix sts=0 sw=2 ts=2 ft=xml: -->  


